File: C#Integration/PokerProtocol/Commands/Game/BetTurnEndedCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;
using PokerProtocol.Commands.Lobby.Response;
using PokerWorld.Game;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Game
{
    public class BetTurnEndedCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "gameBET_TURN_ENDED";

        private readonly TypeRound m_Round;
        private readonly List<int> m_PotsAmounts;

        public TypeRound Round
        {
            get { return m_Round; }
        }

        public List<int> PotsAmounts
        {
            get { return m_PotsAmounts; }
        } 

        public BetTurnEndedCommand(StringTokenizer argsToken)
        {
            int count = int.Parse(argsToken.NextToken());
            m_PotsAmounts = new List<int>();
            for (int i = 0; i < count; ++i)
            {
                m_PotsAmounts.Add(int.Parse(argsToken.NextToken()));
            }
            m_Round = (TypeRound)int.Parse(argsToken.NextToken());
        }

        public BetTurnEndedCommand(List<int> potsAmounts, TypeRound round)
        {
            m_PotsAmounts = potsAmounts;
            m_Round = round;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PotsAmounts.Count);
            foreach( int pa in m_PotsAmounts )
                Append(sb, pa);
            Append(sb, (int)m_Round);
        }
    }
}

File: JAVA/PokerProtocol/src/protocol/commands/game/BetTurnEndedCommand.java
package protocol.commands.game;

import java.util.ArrayList;
import java.util.StringTokenizer;

import poker.game.TypeRound;
import protocol.commands.Command;
import protocol.commands.ICommand;

public class BetTurnEndedCommand implements ICommand
{
    private final ArrayList<Integer> m_potsAmounts;
    private final TypeRound m_round;
    public static String COMMAND_NAME = "gameBET_TURN_ENDED";
    
    public BetTurnEndedCommand(StringTokenizer argsToken)
    {
        final int count = Integer.parseInt(argsToken.nextToken());
        m_potsAmounts = new ArrayList<Integer>();
        for (int i = 0; i < count; ++i)
        {
            m_potsAmounts.add(Integer.parseInt(argsToken.nextToken()));
        }
        m_round = TypeRound.values()[Integer.parseInt(argsToken.nextToken())];
    }
    
    public BetTurnEndedCommand(ArrayList<Integer> potsAmounts, TypeRound round)
    {
        m_potsAmounts = potsAmounts;
        m_round = round;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(BetTurnEndedCommand.COMMAND_NAME);
        sb.append(Command.DELIMITER);
        sb.append(m_potsAmounts.size());
        sb.append(Command.DELIMITER);
        for (int i = 0; i < m_potsAmounts.size(); ++i)
        {
            sb.append(m_potsAmounts.get(i));
            sb.append(Command.DELIMITER);
        }
        sb.append(m_round.ordinal());
        return sb.toString();
    }
    
    public ArrayList<Integer> getPotsAmounts()
    {
        return m_potsAmounts;
    }
    
    public TypeRound getRound()
    {
        return m_round;
    }
}
